R/spread.R

```diff
@@ -94,14 +94,21 @@ spread_.data.frame <- function(data, key_col, value_col, fill = NA,
   if (!is.na(fill)) {
     ordered[is.na(ordered)] <- fill
   }
-  if (convert) {
-    ordered <- type.convert(ordered, as.is = TRUE)
-  }
 
   dim(ordered) <- c(attr(row_id, "n"), attr(col_id, "n"))
   colnames(ordered) <- as.character(col_labels[[1]])
+  ordered <- as.data.frame.matrix(ordered, stringsAsFactors = FALSE)
+
+  if (convert) {
+    ordered[] <- lapply(ordered, type.convert, as.is = TRUE)
+  }
 
-  append_df(row_labels, matrixToDataFrame(ordered))
+  ordered[] <- lapply(ordered, inherit_attributes, value)
+  if (is.factor(value) && drop) {
```

Comment:
You are right, this is now unnecessary and removed.

But in the process of digesting that fact I discovered the need to work a little harder to make `convert = TRUE` work for factor or date input.

So I did that and added related tests and help and example.